packages/agreement-process/src/services/agreementSubmissionProcessor.ts

```diff
@@ -172,11 +173,18 @@ const submitAgreement = async (
         }
       : newAgreement;
 
-  const submittedAgreementEvent = toCreateEventAgreementSubmitted(
-    submittedAgreement,
-    agreementData.metadata.version,
-    correlationId
-  );
+  const agreementEvent =
+    newState === agreementState.active
+      ? toCreateEventAgreementActivated(
+          submittedAgreement,
+          agreementData.metadata.version,
+          correlationId
+        )
+      : toCreateEventAgreementSubmitted(
+          submittedAgreement,
+          agreementData.metadata.version,
+          correlationId
+        );
```

Comment:
The [comment](https://github.com/pagopa/interop-be-monorepo/pull/293#pullrequestreview-2000650963) that opened this issue said "Submission should create an `AgreementActivated` event if the result state of the submission is not Pending" which is slightly different from the logic implemented here (unless the result can only be pending or active). Still, looking at the event name, this makes sense to me - let's wait for @galales check 👍

Reply:
Yes, I maintained the same logic of the scala version. In this case, I think that the only possible values of `newState` are `active` or `pending` so it should be the same.